packages/site/cms-site/src/blocks/factories/SeoBlock.tsx

```diff
@@ -0,0 +1,71 @@
+import Head from "next/head";
+import * as React from "react";
+
+import { PixelImageBlockData, SeoBlockData } from "../../blocks.generated";
+import { withPreview, WithPreviewProps } from "../../iframebridge/withPreview";
+import { generateImageUrl } from "../../image/Image";
+import { PropsWithData } from "../PropsWithData";
+
+type SeoBlockProps<T = PixelImageBlockData> = (T extends PixelImageBlockData
+    ? PropsWithData<SeoBlockData> & { resolveOpenGraphImageUrlTemplate?: never }
+    : { resolveOpenGraphImageUrlTemplate: (block: T) => string | undefined } & PropsWithData<
+          Omit<SeoBlockData, "openGraphImage"> & { openGraphImage: { block?: T } }
+      >) & { title: string; canonicalUrl?: string };
+
+export const SeoBlock: React.FunctionComponent<WithPreviewProps & SeoBlockProps> = withPreview<SeoBlockProps>(
+    <T = PixelImageBlockData,>({
```

Comment:
SeoBlockProps isn't generic here, does this work?

Reply:
Yes because it uses the default type (PixelImageBlockData)

Reply:
Is it possible to to `<SeoBlock<DamImageBlockData> ... />`in the application?

Reply:
Now it is again without `withPreview`

Reply:
data is red because the type of `document.seo` doesn't match DamImageBlockData
<img width="559" alt="image" src="https://github.com/vivid-planet/comet/assets/56400587/06675f07-8cb3-4e48-b91f-88b237fad1b1">